### PR TITLE
docs: point to new "Deno 1.x to 2.x Migration Guide"

### PR DESCRIPTION
**Pull request by iuioiua**
This change reworks deprecation notices to point to the new [Deno 1.x to 2.x Migration Guide](https://docs.deno.com/runtime/manual/advanced/migrate_deprecations).

Prerequisite https://github.com/denoland/deno-docs/pull/300

### CODE REVIEW

**Comment by iuioiua** [comment]
@bartlomieju, please let me know if you like the new format. Once confirmed, I'll apply the change to the rest of the deprecated APIs.

File: runtime/js/90_deno_ns.js [hunk 1]
```diff
@@ -131,7 +131,7 @@ const denoNs = {
     internals.warnOnDeprecatedApi(
       "Deno.read()",
       new Error().stack,
-      "Use `reader.read()` instead.",
+      "See the Deno 1.x to 2.x Migration Guide for migration instructions at https://docs.deno.com/runtime/manual/advanced/migrate_deprecations",
```
Comment (bartlomieju) [inline runtime/js/90_deno_ns.js:134]:
```suggestion
      "See the Deno 1 to 2 Migration Guide for more information at https://docs.deno.com/runtime/manual/advanced/migrate_deprecations",
```

Comment (bartlomieju) [reply runtime/js/90_deno_ns.js:134]: That said - this should be added directly to `warnOnDeprecatedApi` and the current suggestions should be left as they are

**Comment by iuioiua** [comment]
Shall I modify this to only touch JSDocs?

**Comment by bartlomieju** [comment]
> Shall I modify this to only touch JSDocs?

+ update `warnOnDeprecatedApi` to print this message as a last `hint`

File: runtime/js/99_main.js [hunk 2]
```diff
@@ -75,6 +75,7 @@ import {
   workerRuntimeGlobalProperties,
 } from "ext:runtime/98_global_scope_worker.js";
 import { SymbolAsyncDispose, SymbolDispose } from "ext:deno_web/00_infra.js";
+import console from "node:console";
```
Comment (bartlomieju) [inline runtime/js/99_main.js:88]: Revert

**Review by bartlomieju — APPROVED** [review]
Nice work!

**Pull request merged**